test/mocha/20-credentials.js

```diff
@@ -23,674 +23,692 @@ const serviceType = 'vc-issuer';
+          const {data: {verifiableCredential: vc2}} = await zcapClient.write({
+            url: `${sl2021RevocationIssuerId}/credentials/issue`,
+            capability: sl2021RevocationRootZcap,
+            json: {credential: credential2}
+          });
+
+          const vc2StatusId = vc2.credentialStatus.id;
+
+          // this test ensures that the two credentials are not issued with the
+          // same status list index / hash fragment
+          vc1StatusId.should.not.equal(vc2StatusId);
+          const vc1StatusHash = parseInt(vc1StatusId.split('#')[1]);
+          const vc2StatusHash = parseInt(vc2StatusId.split('#')[1]);
+          vc1StatusHash.should.not.equal(vc2StatusHash);
         });
-      } catch(e) {
-        error = e;
-      }
-      assertNoError(error);
-      should.exist(result.data);
-      should.exist(result.data.verifiableCredential);
-      const {verifiableCredential} = result.data;
-      const {proof} = verifiableCredential;
-      should.exist(proof);
-
-      // issue VC with the same ID again (should fail)
-      credential = klona(mockCredential);
-      result = undefined;
-      try {
-        result = await zcapClient.write({
-          url: `${sl2021RevocationIssuerId}/credentials/issue`,
-          capability: sl2021RevocationRootZcap,
-          json: {credential}
+        // ensure duplicate VCs are still properly detected when bookkeeping
+        // fails
+        it('fails when trying to issue a duplicate credential', async () => {
+          const zcapClient = helpers.createZcapClient({capabilityAgent});
+
+          // issue VC (should succeed)
+          let credential = klona(mockCredential);
+          let error;
+          let result;
+          try {
+            result = await zcapClient.write({
+              url: `${sl2021RevocationIssuerId}/credentials/issue`,
+              capability: sl2021RevocationRootZcap,
+              json: {credential}
+            });
+          } catch(e) {
+            error = e;
+          }
+          assertNoError(error);
+          should.exist(result.data);
+          should.exist(result.data.verifiableCredential);
+          const {verifiableCredential} = result.data;
+          const {proof} = verifiableCredential;
+          should.exist(proof);
+
+          // issue VC with the same ID again (should fail)
+          credential = klona(mockCredential);
+          result = undefined;
+          try {
+            result = await zcapClient.write({
+              url: `${sl2021RevocationIssuerId}/credentials/issue`,
+              capability: sl2021RevocationRootZcap,
+              json: {credential}
+            });
+          } catch(e) {
+            error = e;
+          }
+          should.exist(error);
+          error.data.type.should.equal('DuplicateError');
         });
-      } catch(e) {
-        error = e;
-      }
-      should.exist(error);
-      error.data.type.should.equal('DuplicateError');
+      });
     });
-  });
+  }
 });
```

Comment:
This is really hard to review ... I'm going to assume all that happened was that everything was indented here and put into a for loop for each suite. Is that right?

Reply:
Yes, we now test 3 suites: `Ed25519Signature2019`, `Ed25519Signature2020`, and `eddsa-2022`. All are passing in node 16.